test/datasets_utils.py

```diff
@@ -877,7 +877,7 @@ def _make_archive(root, name, *files_or_dirs, opener, adder, remove=True):
     files, dirs = _split_files_or_dirs(root, *files_or_dirs)
 
     with opener(archive) as fh:
-        for file in files:
+        for file in sorted(files):
```

Comment:
@pmeier LMK what you think of this.

Down below I set `buffer_size=1` in the Zipper, because in the original .zip files, both the `image_dp` and the `gt_dp` are fully aligned: they both contain images 00001, 00002, etc. in this order. So I'm assuming that `buffer_size=1` is better than `buffer_size=UNLIMITED`?

Without this call to `sorted()`, the tests would fail: the .zip archive created by `make_zip` would contain the files in a shuffled order (because `files` is a set), and so `image_dp` and the `gt_dp` would not be aligned anymore, leading to a failure to match keys in the Zipper. (Note: this is only a problem in the tests; the code works fine otherwise on my custom script iterating over the dataset).

I hope this won't make other tests fails. This might not be a problem that we have right now, but perhaps something to keep in mind for the future: we might need the test archives to *exactly* match the order of the "original" archives.

Reply:
Good point, I didn't think of that. Do you know if the returned paths of `pathlib.Path.glob()` is stable? If yes, we could simply replace the sets in `_split_files_or_dirs` with lists.

Reply:
I'm not sure about `Path.glob()`. I know that `glob.glob` has no guaranteed order, but I don't think `Path.glob()` relies on it. Maybe the safest is to not assume a specific order.

BTW, slightly related, what was the reason to use sets instead of lists?

Reply:
> what was the reason to use sets instead of lists?

It think the reason was to avoid duplicates, but I don't remember if there was a case where I hit something like that.

Reply:
Have you tried using lists rather than sorting afterwards? If CI is not complaining for other datasets, I feel like that would be the better approach.

Reply:
I just saw this call to `remove()` which might the reason for using sets:

https://github.com/pytorch/vision/blob/afda28accbc79035384952c0359f0e4de8454cb3/test/datasets_utils.py#L862-L863

I can still switch to lists if you'd like, I guess I  would have to write something like

```py
dirs = [dir in dirs if dir != root]
```

LMK which one you prefer